### ISSUE TITLE
have u reproduce the performance of the paper?

### ISSUE THREAD

**Issue by guyafeng** [kayamin/DR-GAN#1]
i have writing a drgan network with tensorflow, but the generated images are not cheerful，then i use ur project, the output is just noise ,so ,i wonder have u reproduce the performance?

**Comment by kayamin** [comment]
Hi, Thank you for trying to run my code and sorry for not working well.
To be honest, I still haven't reproduced the paper performance.
Would you tell me how the generated images are not cheerful ?? I hope we could share how to make the model and our implementation work.

What I tried with single & multi image DRGAN code is

1. train model with our own face image dataset with 7 discrete yaw angles and generate pose conditioned image from the dataset.
2. train model with cfp-dataset with 2 discrete yaw angles and generate pose conditioned image from the cfp-dataset.

In both cases, I could get images which seem to be a face and might preserve identity but could not conditioned their poses i.e yaw angle.
I still don't have an access to Multi-PIE dataset, so could not distinguish this problem whether derives from my poor data quality about pose labeling or my implementation. 

I'm going to try with Multi-PIE dataset within weeks. I'll tell you the result then.

By the way, before updating, my code actually created noise output because I did not applied tanh activation function at the last layer of G_dec. Would you please check up to date codes? 
For now, input images data should be N x 3 x 110 x 110 BGR with [-1,1] intensity. It will be random cropped before feeding them to the model.

**Comment by guyafeng** [comment]
thanks for ur reply, so busy today. I trained drgan with tensorflow using multipie database, the faces are normalized by affine transformation. a nomal dataset acutually help convergence. In my experience, it is  important to condition gans. But model collapse occasionally happend in the later training phase. I will try feature matching and minibatch discrimination these days. 
    And i know the last layer of G should be tanh(), maybe something is wrong when i use ur code, i will check then. 
    Multipie Dataset: i use id、pose and illumination label as in the paper. the pose is between 0-90 degree, I guess the large pose make it harder. different pose will be tested then.
   The generate images will be upload when i know how to create a repertory, LOL!  and if i make some progress i will tell u then.

**Comment by kayamin** [comment]
Hi, I corrected my implementation to apply pose condition properly.
As a result I confirmed that there were some changes on face poses by conditioning their poses on cfp-dataset with a model trained of cfp-dataset.
But still, synthesized image qualities are low.

**Comment by guyafeng** [comment]
Now i can generate different pose faces using a training dataset without illumination, seven poses from nearly 45 to 135 degree.a sample picture is in my reportory: [.](https://github.com/guyafeng/drgan)  what i do is just balance G and D carefully. and no other skills used.
A bit problem: trainingset is about 1000 images, so id is not learned so well.

when i add illumination, images are not so good, still trying...

**Comment by guyafeng** [comment]
Now i can generate different pose faces using a training dataset without illumination, seven poses from nearly 45 to 135 degree.a sample picture is in my reportory [guyafeng/drgan](https://github.com/guyafeng/drgan) what i do is just balance G and D carefully. and no other skills used.
A bit problem: trainingset is about 1000 images, so id is not learned so well.

when i add illumination, images are not so good, still trying...

**Comment by timho102003** [comment]
Hi, Im wondering how do u carefully balance G and D? My G_Loss seems to start converge when reach about 20 epochs and then start diverge with no reason..........
Loss Curve:
https://imgur.com/a/UrLpP
Train Parameters:
Batch Size=64
Train Images=360,000
lr=default
beta1=default
beta2=default

**Comment by ChengXuanhao** [comment]
> 现在我可以使用没有照明的训练数据集生成不同的姿势人脸，从近 45 度到 135 度的七个姿势。我的报告中的示例图片[guyafeng/drgan](https://github.com/guyafeng/drgan)我所做的只是仔细平衡 G 和 D。并且没有使用其他技能。 有点问题：trainingset大概有1000张图片，所以id没学好。
> 
> 当我添加照明时，图像不太好，仍在尝试...

Hello, have you solved this problem? Could you send your code to me for learning?

**Comment by ChengXuanhao** [comment]
> Hi, Thank you for trying to run my code and sorry for not working well. To be honest, I still haven't reproduced the paper performance. Would you tell me how the generated images are not cheerful ?? I hope we could share how to make the model and our implementation work.
> 
> What I tried with single & multi image DRGAN code is
> 
> 1. train model with our own face image dataset with 7 discrete yaw angles and generate pose conditioned image from the dataset.
> 2. train model with cfp-dataset with 2 discrete yaw angles and generate pose conditioned image from the cfp-dataset.
> 
> In both cases, I could get images which seem to be a face and might preserve identity but could not conditioned their poses i.e yaw angle. You can see results sample in cfp-dataset case at demo/Demo_Single_Image_DR_GAN.ipynb. Sorry for in Japanese. I will translate it later. I still don't have an access to Multi-PIE dataset, so could not distinguish this problem whether derives from my poor data quality about pose labeling or my implementation.
> 
> I'm going to try with Multi-PIE dataset within weeks. I'll tell you the result then.
> 
> By the way, before updating, my code actually created noise output because I did not applied tanh activation function at the last layer of G_dec. Would you please check up to date codes? For now, input images data should be N x 3 x 110 x 110 BGR with [-1,1] intensity. It will be random cropped before feeding them to the model.

Hello, have you solved this problem? Can you send your code to me for learning?
